_pytest/nodes.py

```diff
@@ -187,18 +187,34 @@ def find_markers(self, name):
         """find all marks with the given name on the node and its parents
 
         :param str name: name of the marker
-        :returns: iterator over marks matching the name
+        :returns: iterator over marks matching the name"""
+        return map(itemgetter(1), self.find_markers_with_node(name))
+
+    def find_markers_with_node(self, name):
```

Comment:
I find this name confusing TBH... perhaps it would be better to have just one `find_markers` function which always iterates over `(node, mark)` tuples?

Reply:
Another idea: return a namedtuple instead, with `node` and `mark` attributes. It is almost the same, but let's us incrementally add more things to the iteration items if we need in the future.

```
for mark_info in find_markers('skip'):
    if mark_info.node == ... and mark_info.skip.args == ?
```

We just need a better name than `mark_info`.

Reply:
mark_info was always a dead name,

having a "merged" object for mark and node is going to be a major pain, 

the loop example you posted is a unacceptable mush

Reply:
TBH I'm not a big fan of the last example either, just trying to post alternatives. Not sure why you must use "unacceptable mush" to describe it though, because the same loop under your API is:

```python
for (node, mark) in find_markers_with_node('skip'):
    if node.nodeid == ... and mark.args == ...:
```